src/tests/EdFi.DataManagementService.Tests.E2E/Features/References/KeyUnification.feature

```diff
@@ -0,0 +1,139 @@
+                      }
+                  }
+                  """
+             Then the response body is
+                  """
+                  {
+                      "validationErrors":null,
+                      "errors":[
+                          "Constraint failure: document paths $.schoolReference.schoolId and $.sessionReference.schoolId must have the same values"
+                      ],
+                      "detail":"The request could not be processed. See 'errors' for details.",
+                      "type":"urn:ed-fi:api:bad-request",
+                      "title":"Bad Request",
+                      "status":400,
+                      "correlationId":null
+                  }
+                  """
+              And it should respond with 400
+
+        @ignore
+        Scenario: 03 Verify clients cannot update a resource that contains mismatched values on an overlapping natural key field
+            Given Local Course Code "ALG-1 TEST-101"
+            #set value to {id}
+             When a PUT request is made to "ed-fi/courseOfferings/{id}" with
+                  """
+                  {
+                      "localCourseCode": "ALG-1 TEST-101",
+                      "courseReference": {
+                          "courseCode": "ALG-1",
+                          "educationOrganizationId": 255901001
+                      },
+                      "schoolReference": {
+                          "schoolId": 725
+                      },
+                      "sessionReference": {
+                          "schoolId": 999,
+                          "schoolYear": 2022,
+                          "sessionName": "2021-2022 Spring Semester"
+                      }
+                  }
+                  """
+             Then it should respond with 400
+              And the response body is
+                  """
+                  {
+                      "detail": "Data validation failed. See 'validationErrors' for details.",
+                      "type": "urn:ed-fi:api:bad-request:data-validation-failed",
+                      "title": "Data Validation Failed",
+                      "status": 400,
+                      "correlationId": null,
+                      "validationErrors": {
+                          "$.schoolReference.schoolId": [
+                              "All values supplied for 'schoolId' must match. Review all references and align the following conflicting values: '725', '999'"
+                          ],
+                          "$.sessionReference.schoolId": [
+                              "All values supplied for 'schoolId' must match. Review all references and align the following conflicting values: '725', '999'"
+                          ]
+                      }
+                  }
+                  """
+        Scenario: 04 Verify clients can create a resource with a reference to a resource with a complex identity (CourseOffering)
```

Comment:
Test the bug

Reply:
I was confused by this at first, until I realized that the CourseOffering was created above this in another scenario, rather than being created in the background. Now I get it.